frame/contracts/src/wasm/runtime.rs

```diff
@@ -1043,10 +1083,51 @@ define_env!(Env, <E: Ext>,
 		output_len_ptr: u32
 	) -> ReturnCode => {
 		ctx.call(
-			CallFlags::from_bits(flags).ok_or_else(|| "used reserved bit in CallFlags")?,
-			callee_ptr,
-			gas,
-			value_ptr,
+			CallFlags::from_bits(flags).ok_or_else(|| Error::<E::T>::InvalidCallFlags)?,
+			CallType::Call{callee_ptr, value_ptr, gas},
+			input_data_ptr,
+			input_data_len,
+			output_ptr,
+			output_len_ptr,
+		)
+	},
+
+	// Execute code in the context of the current contract.
+	//
+	// The code is executed in the same storage transaction as the caller.
+	// Reentrancy protection is always disabled because you need to trust
+	// the callee anyways.
```

Comment:
Lets improve this doc.

```suggestion
	// Reentrancy protection is always disabled since the callee is allowed
	// to modify the callers storage. This makes going through a reentrancy attack
	// unnecessary for the callee when it wants to exploit the caller.
```